rtl/eclrtl/eclregex.cpp

```diff
@@ -138,6 +138,50 @@ static void failWithPCRE2Error(int errCode, const std::string & msgPrefix, const
 
 //---------------------------------------------------------------------------
 
+// RAII class for PCRE2 8-bit match data
+class PCRE2MatchData8
+{
+private:
+    pcre2_match_data_8 * matchData = nullptr;
+
+public:
+    PCRE2MatchData8() = default;
+    PCRE2MatchData8(pcre2_match_data_8 * newData) : matchData(newData) {}
+    PCRE2MatchData8(const PCRE2MatchData8 & other) : matchData(other.matchData) {}
```

Comment:
Is this correct.  If it was copy assigned I think the item would then be released twice.

Reply:
The RAII pattern actually used in this file is trivial: Capture a new pointer and make sure it is destroyed when it goes out of scope, with no extra copying or movement.  As such, I've removed the copy constructor (and left out things like a move constructor).  The compiler seems happy with leaving the rest of the code as-is.